app_feup/lib/view/Widgets/NavigationDrawer.dart

```diff
@@ -1,70 +1,83 @@
 import 'package:app_feup/view/Theme.dart';
 import 'package:flutter/material.dart';
-
+import '../../utils/Constants.dart' as Constants;
 
 class NavigationDrawer extends StatefulWidget {
   final BuildContext parentContext;
 
-  NavigationDrawer({
-    @required this.parentContext
-  }){}
+  NavigationDrawer({@required this.parentContext}) {}
+
   @override
   State<StatefulWidget> createState() {
     return new NavigationDrawerState(parentContext: parentContext);
   }
 }
 
 class NavigationDrawerState extends State<NavigationDrawer> {
-
   final BuildContext parentContext;
 
-  NavigationDrawerState({
-    @required this.parentContext
-  }){}
+  NavigationDrawerState({@required this.parentContext}) {}
 
-  static final drawerItems = [
-    "Área Pessoal",
-    "Horário",
-    "Mapa de Exames",
-    "Sobre",
-    "Bug Report"
-  ];
+  Map drawerItems = {};
 
-  getCurrentRoute() => ModalRoute.of(parentContext).settings.name == null ? 
-  drawerItems[0]:
-  ModalRoute.of(parentContext).settings.name.substring(1);
+  @override
+  void initState() {
+    super.initState();
+
+    drawerItems = {
+      Constants.NAV_PERSONAL_AREA: _onSelectItem,
+      Constants.NAV_SCHEDULE: _onSelectItem,
+      Constants.NAV_EXAMS: _onSelectItem,
+      Constants.NAV_ABOUT: _onSelectItem,
+      Constants.NAV_BUG_REPORT: _onSelectItem,
+    };
+  }
 
-  _onSelectItem(int index) {
+  // Callback Functions
+  getCurrentRoute() => ModalRoute.of(parentContext).settings.name == null
+      ? drawerItems.keys.toList()[0]
+      : ModalRoute.of(parentContext).settings.name.substring(1);
 
+  _onSelectItem(String key) {
```

Comment:
This method should probably be called onSelectPage, right? If the selected item does not trigger a page change it should not call this method, but instead a method that handles its behavior